src/components/pagination/index.js

```diff
@@ -0,0 +1,130 @@
+import React, { Component, Fragment } from 'react';
+import classNames from 'classnames';
+import { number, bool } from 'prop-types';
+
+export default class Pagination extends Component {
+  static propTypes = {
+    current: number.isRequired,
+    total: number.isRequired,
+    delta: number,
+    showFirst: bool,
+    showLast: bool,
+  };
+  static defaultProps = {
+    delta: 3,
+    showFirst: false,
+    showLast: false,
+  };
+  constructor(props) {
+    super(props);
+    this.state = {
+      ...props,
+      ...this.getPager(this.props.current, this.props.total),
+    };
+  }
+  componentWillReceiveProps = nextProps => {
+    this.setState({
+      ...this.state,
+      ...nextProps,
+      ...this.getPager(nextProps.current, nextProps.total),
+    });
+  };
+  getPager = (current, total) => {
+    const left = current - this.props.delta;
+    const right = current + this.props.delta + 1;
+    const pager = Array.from(
+      {
+        length: total,
+      },
+      (v, k) => k + 1,
+    ).filter(i => i && i >= left && i < right);
+
+    const first = (this.props.showFirst && pager[0] > 1) || false;
+    const last =
+      (this.props.showLast && pager[pager.length - 1] < total) || false;
+
+    return { pager, first, last };
+  };
+  getPrevious = () => {
```

Comment:
For me at least we should try to not have stateful components, as their reusability is reduced. What do you think?

Reply:
I think in a lot of instances this makes sense. For this I wasn't totally sure it would be possible.